File: poa-search-api/src/main/java/nl/rabobank/service/poa/search/detailedpowerofattorney/DetailedPowerOfAttorneys.java
package nl.rabobank.service.poa.search.detailedpowerofattorney;


import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import io.swagger.annotations.ApiModelProperty;
import org.immutables.value.Value;

import javax.annotation.Nonnull;
import java.util.List;

@Value.Immutable
@JsonDeserialize(as = ImmutableDetailedPowerOfAttorneys.class)
public interface DetailedPowerOfAttorneys {
    @Nonnull
    @ApiModelProperty(required = true, value = "Contains all the detailed Power of Attorneys of a client")
    List<DetailedPowerOfAttorney> getDetailedPowerOfAttorneys();
}

File: poa-search-server/src/main/kotlin/nl/rabobank/service/poa/search/detailedpowerofattorney/model/DetailedPowerOfAttorneys.kt
package nl.rabobank.service.poa.search.detailedpowerofattorney.model

import nl.rabobank.service.poa.search.detailedpowerofattorney.ImmutableDetailedPowerOfAttorneys
import nl.rabobank.service.poa.search.util.Exposable
import nl.rabobank.service.poa.search.detailedpowerofattorney.DetailedPowerOfAttorneys as ExposedDetailedPowerOfAttorneys

class DetailedPowerOfAttorneys(
        private val detailedPowerOfAttorneys: List<DetailedPowerOfAttorney>
) : Exposable<ExposedDetailedPowerOfAttorneys> {

    override fun expose(): nl.rabobank.service.poa.search.detailedpowerofattorney.DetailedPowerOfAttorneys =
            ImmutableDetailedPowerOfAttorneys.builder()
                    .addAllDetailedPowerOfAttorneys(this.detailedPowerOfAttorneys.map { it.expose() })
                    .build()
}